gensim/sklearn_integration/SklearnWrapperGensimLdaModel.py

```diff
@@ -0,0 +1,116 @@
+        returns all parameters as dictionary.
+        Warnings: Must for sklearn API.Do not Remove.
+        """
+        if deep:
+            return {"corpus":self.corpus,"num_topics":self.num_topics,"id2word":self.id2word,
+                    "distributed":self.distributed,"chunksize":self.chunksize,"passes":self.passes,
+                    "update_every":self.update_every,"alpha":self.alpha," eta":self.eta," decay":self.decay,
+                    "offset":self.offset,"eval_every":self.eval_every," iterations":self.iterations,
+                    "gamma_threshold":self.gamma_threshold,"minimum_probability":self.minimum_probability,
+                    "random_state":self.random_state}
+
+    def set_params(self, **parameters):
+        """
+        set all parameters.
+        Warnings: Must for sklearn API.Do not Remove.
+        """
+        for parameter, value in parameters.items():
+            self.setattr(parameter, value)
+        return self
+
+    def fit(self, X):
+        """
+        For fitting corpus into the class object.
+        calls gensim.model.LdaModel:
+        >>>gensim.models.LdaModel(corpus=corpus,num_topics=num_topics,id2word=id2word,passes=passes,update_every=update_every,alpha=alpha,iterations=iterations,eta=eta,random_state=random_state)
+        Warnings: Must for sklearn API. Do not Remove.
+        """
+        self.corpus=X
+        models.LdaModel.__init__(
+                                 self, corpus=X, num_topics=self.num_topics, id2word=self.id2word,
+                                 distributed=self.distributed, chunksize=self.chunksize, passes=self.passes,
+                                 update_every=self.update_every,alpha=self.alpha, eta=self.eta, decay=self.decay,
+                                 offset=self.offset,eval_every=self.eval_every, iterations=self.iterations,
+                                 gamma_threshold=self.gamma_threshold,minimum_probability=self.minimum_probability,
+                                 random_state=self.random_state)
+        return self
+
+    def transform(self, bow, minimum_probability=None, minimum_phi_value=None, per_word_topics=False):
+        """
+        takes as an input a new document (bow) and
+        Return topic distribution for the given document bow, as a list of (topic_id, topic_probability) 2-tuples.
+        Warnings: Must for sklearn API. Do not Remove.
+        """
+        return self.get_document_topics(
+                                        bow, minimum_probability=minimum_probability,
+                                        minimum_phi_value=minimum_phi_value, per_word_topics=per_word_topics)
+
+    def partial_fit(self, X):
+        """
+        train model over X.
+        """
+        self.update(corpus=X)
+
+    def fit_predict(self,X):
+        """
+        predict topics for given numpy array.
+        Warnings: Important for sklearn Pipeline API. Do not Remove.
+        """
+        corpus = matutils.Sparse2Corpus(X)
```

Comment:
Move this line to the fit function. Add an If to only run this line if necessary. Add a test for this functionality